pkg/util/json/json.go

```diff
@@ -656,21 +656,49 @@ func (j jsonNumber) EncodeInvertedIndexKeys(b []byte) ([][]byte, error) {
 }
 func (j jsonArray) EncodeInvertedIndexKeys(b []byte) ([][]byte, error) {
 	var outKeys [][]byte
+	var prevChildren = make([][]byte, 0)
 
 	for i := range j {
 		children, err := j[i].EncodeInvertedIndexKeys(nil)
 		if err != nil {
 			return nil, err
 		}
+		children = removeOverlappingKeys(prevChildren, children)
 		for _, childBytes := range children {
+			if childBytes == nil {
+				continue
+			}
 			encodedKey := bytes.Join([][]byte{b, encoding.EncodeArrayAscending(nil), childBytes}, nil)
 			outKeys = append(outKeys, encodedKey)
 		}
+		prevChildren = children
 	}
 
 	return outKeys, nil
 }
 
+func removeOverlappingKeys(keys1 [][]byte, keys2 [][]byte) [][]byte {
```

Comment:
It will be very valuable to understand what this function is meant to do. What are the constraints on the parameters and what is returned.